### PR TITLE
Workaround for extremely long image decode

### PR DESCRIPTION
**Pull request by FooIbar**
Workaround for #732

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/image/Image.kt [hunk 1]
```diff
@@ -112,9 +114,7 @@ class Image private constructor(image: CoilImage, private val src: AutoCloseable
                 if (isAtLeastO) {
                     colorSpace(colorSpace)
                 }
-                if (Settings.cropBorder.value) {
-                    allowHardware(false)
-                }
+                allowHardware(false)
```
Comment (VDostoyevskiy) [inline app/src/main/kotlin/com/hippo/ehviewer/image/Image.kt:117]: For non-crop scenarios, we always use a non-hw bitmap, which decreases performance

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/image/Image.kt:117]: Well, we have no information about image size so can't disable hardware bitmap based on that.

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/image/Image.kt:117]: > Well, we have no information about image size so can't disable hardware bitmap based on that.

Make non-hw decode a fallback path

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/image/Image.kt:117]: > For non-crop scenarios, we always use a non-hw bitmap, which decreases performance

Wait didn't we copy it to hardware bitmap regardless of crop borders settings?

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/image/Image.kt:117]: > > For non-crop scenarios, we always use a non-hw bitmap, which decreases performance
> 
> Wait didn't we copy it to hardware bitmap regardless of crop borders settings?

Oops, that's terrible, we need to fix it up

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/image/Image.kt:117]: Let's add a Interceptor to address the fallback path
check hardware flags first, if true, check the result, if is that Exception then we re-decode it

**Comment by VDostoyevskiy** [comment]
Merged into https://github.com/FooIbar/EhViewer/pull/791

**Pull request closed without merging**